### PR TITLE
Activate operators metadata uploader plugin

### PR DESCRIPTION
**Pull request by athos-ribeiro**

### CODE REVIEW

File: osbs/api.py [hunk 1]
```diff
@@ -614,6 +614,7 @@ def _do_create_prod_build(self, git_uri, git_ref,
                               reactor_config_override=None,
                               parent_images_digests=None,
                               git_commit_depth=None,
+                              allowed_platform=None,
```
Comment (twaugh) [inline osbs/api.py]: I can't think of a better name for this :-(

File: inputs/worker_inner:6.json [hunk 2]
```diff
@@ -83,6 +83,9 @@
         "publish": false
       }
     },
+    {
+      "name": "export_operator_manifests"
```
Comment (twaugh) [inline inputs/worker_inner:6.json]: Needs test-case change.

File: osbs/api.py (hunk 1 above)
Comment (athos-ribeiro) [reply osbs/api.py]: maybe `operator_uploader` or `operator_uploader_platform`?

Comment (twaugh) [reply osbs/api.py]: The best I came up with was `noarch_platform`, i.e. the platform whose worker can perform operations for which it doesn't matter which platform is used. But the fact that I have to explain it probably means it's a bad name.

I prefer either of `operator_uploader` or `operator_uploader_platform` to `allowed_platform`.

Comment (twaugh) [reply osbs/api.py]: Or even `any_platform`.

Comment (MartinBasti) [reply osbs/api.py]: I vote for `operator_uploader_platform`, that names describes what is that param used for

Comment (MartinBasti) [reply osbs/api.py]: BTW, in design doc we agreed on `operator_manifests_extract_platform`

File: osbs/build/build_request.py [hunk 3]
```diff
@@ -99,6 +99,7 @@ def set_params(self, **kwargs):
         :param auto_build_node_selector: dict, a nodeselector for auto builds
         :param isolated_build_node_selector: dict, a nodeselector for isolated builds
         :param is_auto: bool, indicates if build is auto build
+        :param allowed_platform: str, indicates which platform should upload operator manifests
```
Comment (mlangsdorf) [inline osbs/build/build_request.py]: is this a valid parameter to a v1 build_request?  You can't run the operators metadata plugin in any of the arrangements supported by v1 build_request.

File: osbs/build/user_params.py [hunk 4]
```diff
@@ -125,6 +126,7 @@ def set_params(self,
                 self.build_imagestream.value = source_value
 
         self.parent_images_digests.value = parent_images_digests
+        self.allowed_platform.value = allowed_platform
```
Comment (mlangsdorf) [inline osbs/build/user_params.py]: you'll also need to add allowed_platform to atomic-reactor's user_params schema.

Comment (athos-ribeiro) [reply osbs/build/user_params.py]: Done in https://github.com/projectatomic/atomic-reactor/pull/1177
Thanks for catching this one :)

File: osbs/build/build_request.py (hunk 3 above)
Comment (athos-ribeiro) [reply osbs/build/build_request.py]: Removed

File: osbs/utils.py [hunk 5]
```diff
@@ -606,7 +608,8 @@ class Labels(object):
         LABEL_TYPE_HOST: ('com.redhat.build-host', 'Build_Host'),
         LABEL_TYPE_RUN: ('run', 'RUN'),
         LABEL_TYPE_INSTALL: ('install', 'INSTALL'),
-        LABEL_TYPE_UNINSTALL: ('uninstall', 'UNINSTALL')
+        LABEL_TYPE_UNINSTALL: ('uninstall', 'UNINSTALL'),
+        LABEL_TYPE_OPERATOR_MANIFESTS: ('com.redhat.delivery.appregistry',)
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/utils.py]: generally we using it for labels which have variants, but we also use this, as only place where we define label names, so this way if in the future there will be added new name we change it just here and we are done

**Comment by athos-ribeiro** [comment]
Release notes added:

## Build JSON changes
- add operator_manifests_extract_platform parameter to indicate which platform should upload operator manifests to koji

## Improvements
- Run new export_operator_manifests plugin in worker builds to upload files
- Add support for `com.redhat.delivery.appregistry` label in Dockerfile. When set to 'true', this label indicates the presence of operator manifests in the built image

**Pull request merged**